Feature/cdp 58 add authentication service to FrontEnd

OfekSagiv opened the pull request:
- Implemented a modular authentication service with login and logout methods using Axios.  
- Configured a singleton Axios instance with token-based authorization and 401 error handling.  
- All constants, messages, and endpoints are externalized.  
- The service is ready for use in any login-related UI components.

OfekSagiv commented:
@Tamir198 @noagedo  
Ready for your feedback.

Tamir198 commented on src/api/axios.js:
```diff
@@ -0,0 +1,13 @@
+import { STORAGE_KEYS, ROUTES } from '../constants/auth.constants';
+
+const getToken = () => localStorage.getItem(STORAGE_KEYS.TOKEN);
+
```
Local/session storage should be saved into a service, the service should expose get and set methods, so in the future if we want to add something to it we will do it in one place

Tamir198 commented on src/api/axiosInstanceFactory.js:
```diff
@@ -0,0 +1,35 @@
+export const createAxiosInstance = ({ getToken, onUnauthorized }) => {
+  const instance = axios.create({
+    baseURL: import.meta.env.VITE_API_URL,
+    withCredentials: true,
```
Maybe instead of `VITE_API_URL` call this `BASE_SERVER_URL` or anything that indicates what the variable is doing. 

Also please update the example env to include the new key

Tamir198 commented on src/api/axiosInstanceFactory.js:
```diff
@@ -0,0 +1,35 @@
+  instance.interceptors.request.use(
+    (config) => {
+      const token = getToken?.();
+      if (token) {
```
Do you think that the `?` part is necessary ?
Because we are creating 1 instance of axios (and this is when the app start) 
We can see if there is something wrong when the app starts for the first time

Tamir198 commented on src/api/axiosInstanceFactory.js:
```diff
@@ -0,0 +1,35 @@
+    (response) => response,
+    (error) => {
+      const status = error.response?.status;
+
```
Same question about `?`

OfekSagiv replied:
getToken should always be defined, so optional chaining isn’t needed there and I’ve removed it.
In contrast, error.response?.status is kept intentionally since response may be undefined in some error cases (e.g., network failure).

Tamir198 replied:
ok

Tamir198 commented on src/services/authService.js:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    localStorage.setItem(STORAGE_KEYS.TOKEN, token);
+    return { token, user };
```
This should be replaced with the storage session that you will create

Tamir198 commented on src/services/authService.js:
```diff
@@ -0,0 +1,70 @@
+    const message = error.response?.data?.message;
+
+    if (status === HTTP_STATUS.UNAUTHORIZED) {
+      throw new Error(AUTH_ERRORS.UNAUTHORIZED_LOGIN);
```
This should come from the status codes library

Tamir198 commented on src/services/authService.js:
```diff
@@ -0,0 +1,70 @@
+    const success = response.data?.[AUTH_RESPONSE_KEYS.SUCCESS];
+    if (!success) {
+      throw new Error(AUTH_ERRORS.LOGOUT_FAILED);
+    }
```
Status codes from library in here as well

OfekSagiv replied:
This constant is used as an error message, while the status code check itself is already handled above using StatusCodes.UNAUTHORIZED. are you agreeing with that ? this continues for the next comments below.

OfekSagiv replied:
Maybe make a custom error object? 
throw new CustomError(AUTH_ERRORS.UNAUTHORIZED_LOGIN, StatusCodes.UNAUTHORIZED);
?

Tamir198 replied:
ye that's fine, for custom error you can use this library :

https://www.npmjs.com/package/http-errors

Or create an error service- you decide

Tamir198 replied:
Just please talk with the team about this change so they will know what do to and how to handle errors in the front end

Tamir198 reviewed (changes requested):
Left you some comments, tag me for re review once you will finish with them

OfekSagiv commented:
@Tamir198 
Answer here on the second comment, environment variables must be prefixed with VITE_ to be accessible in the frontend. I’ve updated the .env and usage accordingly. without prefix you can't get the backend with API/V1... prefix. checked this issue with dummy page for login to ensure everything works. so I changed it to :
  baseURL: import.meta.env.VITE_BASE_SERVER_URL,

OfekSagiv commented:
@Tamir198 Waiting for your re-review

Tamir198 reviewed (changes requested):
Overall looks better, just one thing: 

You have wrote 2 services using 2 different approach: 

1) static methods (object) 
2) named export (export const)

Any reason for the differences?

OfekSagiv commented:
I agree with you and changed to object approach @Tamir198
